File: transpositionEncrypt.py
# Transposition Cipher Encryption Program
# BSD Licensed

import pyperclip

def main():
  myMessage = 'Common sense is not so common.'
  mykey = 8

  chifertext = encryptMessage(mykey, myMessage)
  print('Ciphertext:')
  print(chifertext + '|')
  pyperclip.copy(chifertext)
  print('Ciphertext copied to clipboard.')

def encryptMessage(key, message):
  ciphertext = [''] * key

  for column in range(key):
    currentIndex = column

    while currentIndex < len(message):
      ciphertext[column] += message[currentIndex]

      currentIndex += key

  return ''.join(ciphertext)

if __name__ == '__main__':
  main()

File: transpositionEncrypt.rb
# Transposition Cipher Encryption Program
# BSD Licensed

require_relative 'pyperclip'

def main()
  myMessage = 'Common sense is not so common.'
  mykey = 8

  ciphertext = transpositionEncrypt(myMessage, mykey)

  puts "Ciphertext: #{ciphertext} |"
  RubyClip.copy(ciphertext)
  puts "Ciphertext copied to clipboard."
end

def transpositionEncrypt(message, key)
  ciphertext = [''] * key

  message.chars.each_with_index do |char, index|
    column = index % key
    ciphertext[column] += char
  end

  ciphertext.join
end

if __FILE__ == $0
  main()
end
